domain/model/service/modelservice.go

```diff
@@ -5,16 +5,26 @@ package service
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/juju/clock"
 
+	coreerrors "github.com/juju/juju/core/errors"
 	coremodel "github.com/juju/juju/core/model"
 	corestatus "github.com/juju/juju/core/status"
 	"github.com/juju/juju/domain/model"
+	"github.com/juju/juju/environs"
 	"github.com/juju/juju/internal/errors"
 	"github.com/juju/juju/internal/uuid"
 )
 
+// ModelEnvironProvider is the model environ provider required by this service.
+type ModelEnvironProvider interface {
```

Comment:
I would just call this `EnvironProvider`. I can't see a reason to have model prefixed here as model's don't form part of this construct.

Reply:
Sure